### PR TITLE
exfatprogs: Fix -Wsign-compare warnings

### PR DESCRIPTION
**Pull request by luca020400**

### CODE REVIEW

**Comment by luca020400** [comment]
With this clang-10 is happy with -Werror -Wall -Wextra :)

**Comment by namjaejeon** [comment]
@luca020400 If you add your signed-off-by in patch, I will be happy!

**Comment by namjaejeon** [comment]
> With this clang-10 is happy with -Werror -Wall -Wextra :)

This description should be in patch header ?

**Comment by luca020400** [comment]
Done

File: fsck/fsck.c [hunk 1]
```diff
@@ -327,7 +327,7 @@ static ssize_t exfat_file_read(struct exfat *exfat, struct exfat_inode *node,
 	ssize_t read_size;
 	size_t remain_size;
 
-	if (file_offset >= node->size)
+	if (file_offset >= (off_t)node->size)
```
Comment (hclee) [inline fsck/fsck.c:330]: Can you change the type of size to off_t?

File: fsck/fsck.c [hunk 2]
```diff
@@ -383,7 +383,7 @@ static int boot_region_checksum(struct exfat *exfat)
 	__le32 checksum;
 	unsigned short size;
 	void *sect;
-	int i;
+	unsigned long i;
```
Comment (hclee) [inline fsck/fsck.c:386]: Why is long type needed instead of int?

**Comment by luca020400** [comment]
Please take a look!
Replaced unsigned long with size_t when used to compare sizeof()

File: fsck/fsck.c [hunk 3]
```diff
@@ -383,7 +383,7 @@ static int boot_region_checksum(struct exfat *exfat)
 	__le32 checksum;
 	unsigned short size;
 	void *sect;
-	int i;
+	size_t i;
```
Comment (hclee) [inline fsck/fsck.c:386]: if int is replaced with unsigned int, warnings are generated? size_t is also unsigned long in some system.

**Comment by hclee** [comment]
@luca020400, this PR is merged into exfat-next with PR #96 . Can I close this PR?

**Pull request closed without merging**